pkg/dataplane/client.go

```diff
@@ -60,7 +62,30 @@ func NewClient(aud, cloud string, cred azcore.TokenCredential) (*ManagedIdentity
 		},
 	}
 
-	azCoreClient, err := azcore.NewClient("managedidentitydataplane.APIClient", moduleVersion, plOpts, nil)
+	azCoreClient, err := azcore.NewClient(moduleName, moduleVersion, plOpts, nil)
+	if err != nil {
+		return nil, err
+	}
+	swaggerClient := swagger.NewSwaggerClient(azCoreClient)
+
+	return &ManagedIdentityClient{swaggerClient: swaggerClient, cloud: cloud}, nil
+}
+
+// NewStubClient creates a stubbed Managed Identity Dataplane API client for testing purposes
+func NewStubClient(cloud string, transport *stub) (*ManagedIdentityClient, error) {
```

Comment:
you can use the api pattern used in azure sdk for go, pass the optional transport via the client options. Then you won't need this stub function.

Reply:
The pipeline options differ since we don't have bearer challenge implemented in the stub. We could make this is a function param, but I think it's simpler to just have two separate functions.

Reply:
where will this stub client be used?

Reply:
I suggest taking a look how fleet is doing this. I think that's cleaner. having a stub function isn't ideal.

https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=%2Ffleet%2Fpkg%2Fapi%2Foperations%2Ffleet%2Fput_fleet_test.go&_a=contents&version=GBmaster

Reply:
It will be consumed by ARO classic RP and hcp RP. So for local development, those RPs can consume the stub to mock MSI-RP behavior.

I'll refactor this code following the fleet suggestion so that both RPs can just call one function and use a fake transpor.

Reply:
Refactored NewClient() so we no longer have separate functions for stub client creation.